bokeh/charts/builder.py

```diff
@@ -547,6 +552,33 @@ def generate_help(cls):
 
         return help_str
 
+    @staticmethod
+    def _sort_legend(sort_legend, legends, attributes):
+        """Sort legends sorted by looping though sort_legend items (
+        see :attr:`Builder.sort_legend` for more details)
+        """
+        if len(sort_legend) > 0:
+            for attr, asc in sort_legend:
+                if len(attributes[attr].columns) > 0:
+
+                    # TODO(fpliger): attributes should be consistent and not
+                    #               need any type checking but for
+                    #               the moment it is not, specially when going
+                    #               though a process like binning or when data
+                    #               is built for HeatMap, Scatter, etc...
+                    item_order = [x[0] if isinstance(x, tuple) else x
+                        for x in attributes[attr].items]
+
+                    item_order = [str(x) if not isinstance(x, string_types)
+                        else x for x in item_order]
+
+                    def foo(leg):
+                        return item_order.index(leg[0])
+
+                    return list(sorted(legends, key=foo, reverse=(not asc)))
```

Comment:
@bryevdv thinking a bit more about this I'm not really sure it makes sense since all attributes would refer to the same legends so wouldn't be sub-sorting but re-sorting instead which would not be correct. I can't find a use case when you'd use more then one attribute in `sort_legend`. It would make more sense if `sort_legend` was a `Dict` or a `Tuple(String, Bool)`...